client/lib/signup/step-actions/index.js

```diff
@@ -318,6 +343,17 @@ export function createSiteWithCart( callback, dependencies, stepData, reduxStore
 				isFreeThemePreselected,
 				themeSlugWithRepo
 			);
+			recordSignupComplete( {
```

Comment:
In the [function definition](https://github.com/Automattic/wp-calypso/blob/6b3c29ae140352b47ba3f8bc2c5ed41b5b5b34d6/client/lib/analytics/signup.js#L27) of `recordSignupComplete()`, I feel we should decouple the Tracks calls to `calypso_signup_complete` and `calypso_new_user_site_creation`. In other words, we should have a separate function to record `calypso_new_user_site_creation` instead of bundling it in the same function as `recordSignupComplete()`.

The reason for this is that `calypso_signup_complete` records the completion of a signup flow. We have many signup flows that don't lead to site creation, for example the `https://wordpress.com/start/account` flow or `/start/domain` flow. Per this PR's change to record these events as a response handler to `/sites/new`, the signup complete event will not get recorded in these flows. 

I think we should continue to keep the changes in `client/signup/main.jsx` file, but remove `calypso_new_user_site_creation` from `recordSignupComplete()`, and then record site creation alone in the /sites/new callback. Does this make sense to you?